### PR TITLE
mgr/dashboard: Allow "/" in pool name

### PR DESCRIPTION
**Pull request by tspmelo**
Fixes: http://tracker.ceph.com/issues/38302

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by votdev** [comment]
After creating a pool ``ceph osd pool create my/pool 8`` i was able to open the edit/form dialog. But after changing the name to ``my/pool_test`` i got an error message after pressing ``Edit pool``.

failure

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ export class PoolFormComponent implements OnInit {
     this.form = new CdFormGroup(
       {
         name: new FormControl('', {
-          validators: [Validators.pattern('[\\.A-Za-z0-9_-]+'), Validators.required]
+          validators: [Validators.pattern(/[\.A-Za-z0-9_\-/]+/), Validators.required]
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts]: The ``-`` does not have to escaped (https://javascript.info/regexp-escaping). I think to remember that it is common practice to put the ``-`` to the end char list.

**Comment by tspmelo** [comment]
@votdev Fixed.

**Comment by ricardoasmarques** [comment]
I've updated the corresponding issues with new findings...

But in summary, if I try to create an image in a pool that contains a slash ('/') in the name, I'll get the following error:

    # rbd create --size 1G --pool my/pool --image image
    rbd: invalid pool name 'my/pool'

So we should not allow the user to create pools with a '/' in the name and application "rbd".

And we should not allow the user to create an image with a '/' in the name, or in a pool with a '/' in the name (that may have been created from the command line).

**Comment by votdev** [comment]
> I've updated the corresponding issues with new findings...
> 
> But in summary, if I try to create an image in a pool that contains a slash ('/') in the name, I'll get the following error:
> 
> ```
> # rbd create --size 1G --pool my/pool --image image
> rbd: invalid pool name 'my/pool'
> ```
> 
> So we should not allow the user to create pools with a '/' in the name and application "rbd".
> 
> And we should not allow the user to create an image with a '/' in the name, or in a pool with a '/' in the name (that may have been created from the command line).

Is there any Ceph documentation that is clarifying if it is allowed or not to use ``/`` in names?

**Comment by ricardoasmarques** [comment]
> Is there any Ceph documentation that is clarifying if it is allowed or not to use `/` in names?

I don't know about the documentation, but the validation regex is here - 
https://github.com/ceph/ceph/blob/master/src/tools/rbd/Utils.cc#L347

The inconsistency here is that you can create it if you use the python bindings (e.g. by using the dashboard), but will not be allowed if using the command line.

**Comment by tspmelo** [comment]
This PR now focus only on the pool form.
It allows `/` only when `rbd` is not selected.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html [hunk 2]
```diff
@@ -425,8 +430,18 @@ <h3 class="panel-title">
...
 
             </div>
-          </div>
 
+            <div class="form-group has-error">
+              <div class="col-sm-offset-3 col-sm-9"
+                   *ngIf="form.hasError('rbdPool')">
+                <br>
+                <span class="help-block"
+                      i18n>It's not possible to create an RBD pool with '/' in the name.</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html]:
```suggestion
                      i18n>It's not possible to create a RBD pool with '/' in the name.</span>
```

Comment (LenzGr) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html]: Actually, "an RBD pool" is correct: https://www.ecenglish.com/learnenglish/when-use-a

**Pull request merged**